libsplinter/src/service/timer_filter.rs

```diff
@@ -14,9 +14,9 @@
 
 use crate::error::InternalError;
 
-use super::FullyQualifiedServiceId;
+use super::{FullyQualifiedServiceId, Typed};
 
-pub trait TimerFilter {
+pub trait TimerFilter: Typed {
```

Comment:
We should discuss whether TimerFilter should be support more than a single service type. For example, if we have "scabbard::v3" and "scabbard::v4", should we require different TimerFilter instances. This is also going to be the same for MessageHandler, etc.

Reply:
Also Typed should return a struct like ServiceType - should discuss whether that change should go in before or after this.